nexus/reconfigurator/planning/tests/output/planner_nonprovisionable_1_2.txt

```diff
@@ -1,89 +1,137 @@
-     nexus          15c103f0-ac63-423b-ba5d-1b5fcd563ba3   in service    fd00:1122:3344:101::22         
-+    nexus          3ca5292f-8a59-4475-bb72-0f43714d0fff   in service    fd00:1122:3344:101::2e  added  
-+    nexus          99f6d544-8599-4e2b-a55a-82d9e0034662   in service    fd00:1122:3344:101::2d  added  
-+    nexus          c26b3bda-5561-44a1-a69f-22103fe209a1   in service    fd00:1122:3344:101::2f  added  
+  --------------------------------------------------------------------------------------------------------
+     zone type      zone ID                                disposition   underlay IP             status   
+  --------------------------------------------------------------------------------------------------------
+                                                                                                          
+  UNCHANGED SLEDS:                                                                                        
+                                                                                                          
+   sled 2d1cb4f2-cf44-40fc-b118-85036eb732a9: zones at generation 2                                       
+     crucible       19fbc4f8-a683-4f22-8f5a-e74782b935be   in service    fd00:1122:3344:105::26           
+     crucible       4f1ce8a2-d3a5-4a38-be4c-9817de52db37   in service    fd00:1122:3344:105::2c           
+     crucible       6b53ab2e-d98c-485f-87a3-4d5df595390f   in service    fd00:1122:3344:105::27           
+     crucible       93b137a1-a1d6-4b5b-b2cb-21a9f11e2883   in service    fd00:1122:3344:105::23           
+     crucible       9f0abbad-dbd3-4d43-9675-78092217ffd9   in service    fd00:1122:3344:105::25           
+     crucible       b0c63f48-01ea-4aae-bb26-fb0dd59d1662   in service    fd00:1122:3344:105::28           
+     crucible       c406da50-34b9-4bb4-a460-8f49875d2a6a   in service    fd00:1122:3344:105::24           
+     crucible       d660d7ed-28c0-45ae-9ace-dc3ecf7e8786   in service    fd00:1122:3344:105::2a           
+     crucible       e98cc0de-abf6-4da4-a20d-d05c7a9bb1d7   in service    fd00:1122:3344:105::2b           
+     crucible       f55e6aaf-e8fc-4913-9e3c-8cd1bd4bdad3   in service    fd00:1122:3344:105::29           
+     internal_ntp   7f4e9f9f-08f8-4d14-885d-e977c05525ad   in service    fd00:1122:3344:105::21           
+     nexus          6dff7633-66bb-4924-a6ff-2c896e66964b   in service    fd00:1122:3344:105::22           
+                                                                                                          
+  MODIFIED SLEDS:                                                                                         
+                                                                                                          
+*  sled 48d95fef-bc9f-4f50-9a53-1e075836291d: zones at generation: 2 -> 3                                 
```

Comment:
pretty nitty and maybe not worth doing anything about now: it's a little weird to me that we go from generation 2 -> 3 to mark all the zones expunged.  I think that's because I think of this as the generation number of the list of zones that we tell the sled agent about.  When the sled is expunged, that list is non-existent.  Arguably we could say that's generation 3 and empty -- that'd be fine -- but it wouldn't contain expunged zones.

On the other hand, with the splitting of inventory zones vs. blueprint zones, I guess expunged zones may be in the blueprint but not the inventory or the set that we send the sled agent.  So maybe the generation number here covers not the list that we send to the sled agent, but the blueprint list.  And that does include expunged zones.  And then the list of zones that we would send to the sled is a filter over the blueprint list?  If that's how we're thinking of this (which seems reasonable), then I think I might have this output say "blueprint zones at generation: 2 -> 3" instead of just "zones at generation: 2 -> 3".

It might also be worth annotating the sled row to say that the sled itself is expunged.

Reply:
Yeah, re this and the next one -- we're currently only storing the fact that the zone or sled is expunged at the moment. We aren't storing information about why the zone or sled was expunged in any persistent form. While writing this I started encountering the issue you did as well -- a sled or a bunch of zones being marked expunged without that context can be confusing.

I'd like to revisit this after release 8, potentially adding a comment field to the database. Is that okay?

Reply:
Sounds good.  Are you okay with adding the word "blueprint" to the "zones at generation", just to clarify that we're now talking about the blueprint zones and not the set we're sending to the sled agent?

Reply:
Sure, that makes sense.

Reply:
Done.